C-Sharp/sort/MergeSort.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Merge sort is an efficient, general-purpose, comparison-based sorting algorithm.
+// Implementation of merge sort
```

Comment:
Add a better description

Reply:
Done

Reply:
I only changed the comments.  but I got this error in checks "DeepSource: Java (beta) Failing after 7s — Analysis Failed: Issues detected that need attention."